### DISCUSSION TITLE
Help with installing. Uses wrong node version maybe.

### DISCUSSION THREAD

**Discussion by jakkarn** [neoclide/coc.nvim#4603] [Q&A] [answered]
Hi, I've tried to install coc.nvim, and the PlugInstall command succeeds, but when I open up nvim again I get an error `build/index.js not found, please install dependencies and compile coc.nvim by: yarn install`. I looked at dependencies and made sure to install node.js (and yarn as well after no success). I run PlugClean after commenting out the Plug line in init.vim and then PlugInstall after uncommenting it again.

I have tried these three versions of the Plug config:
```
Plug 'neoclide/coc.nvim' ", {'branch': 'release'}
Plug 'neoclide/coc.nvim', {'branch': 'master', 'do': 'yarn install --frozen-lockfile'}
Plug 'neoclide/coc.nvim' ", {'rev': 'release'}
```
When I ran the installation with `'do': 'yarn install --frozen-lockfile'` I got an error `Post-update hook for coc.nvim ... Exit status: 1` which I understand as the yarn install failing, but no idea why.

At this point I just winged it and tried going to the `~/.local/share/nvim/plugged/coc.nvim` directory and manually ran `yarn install --frozen-lockfile`. Not sure what to expect, I got the following output:
```
[1/5] Validating package.json...
error coc.nvim-master@0.0.82: The engine "node" is incompatible with this module. Expected version ">=14.14.0". Got "8.16.0"
error Found incompatible module.
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
```
So it seems that coc.nvim sees the wrong version of node because when I run `node --version` I can see version 18.15.0.

I also already added `let g:coc_node_path = '/usr/local/bin/node'` to init.vim already, which is the node binary with the correct version.

Any help? :)

**Comment by fannheyward** [comment]
It's recommended to use `Plug 'neoclide/coc.nvim', {'branch': 'release'}`.

> error coc.nvim-master@0.0.82: The engine "node" is incompatible with this module. Expected version ">=14.14.0". Got "8.16.0"

This error is came from `yarn`, the node used by yarn is version 8.16.0, outdated, looks like `yarn` didn't use your v18.15.0 node.

**Reply by jakkarn** [reply]
Thanks for the quick response. Yarn isn't included as a dependency for the {'branch': 'release'} installation (unless I'm mistaken). Do you think that can have anything to do with why CoC isn't installed properly in the standard case?

**Reply by fannheyward** [reply]
Yes, you don't need `yarn` with `release` branch. Upgrade your vim-plug and try again, looks like vim-plug failed to clone branch.

**Reply by jakkarn** [reply]
I've run `PlugUpgrade` from vim with "already up to date" as result. Removing and installing the release branch again resulted in same problem. The `PlugInstall` command outputs that it finished installing successfully. Still get the same `build/index.js not found...` when I start vim.

**Reply by jakkarn** [reply, marked as answer]
I managed to get it to work now, but only with the master branch and with the yarn install flags. I realized that yarn looked at some other node version, so I removed some node binaries I found and removed yarn. Then I installed only yarn again with npm. I think I might have installed an old version of node-js via yarn or npm or something. The release branch still didn't work though. Would be nice to use the release version.

Thanks for the good work on Conqueror of Completion.
